aviary/mission/gasp_based/phases/descent_phase.py

```diff
@@ -1,11 +1,243 @@
 from aviary.mission.gasp_based.ode.descent_ode import DescentODE
 from aviary.mission.phase_builder_base import PhaseBuilderBase
 from aviary.mission.initial_guess_builders import InitialGuessState, InitialGuessIntegrationVariable, InitialGuessControl
+from aviary.utils.aviary_options_dict import AviaryOptionsDictionary
 from aviary.utils.aviary_values import AviaryValues
 from aviary.variable_info.enums import SpeedType
 from aviary.variable_info.variables import Dynamic
 
 
+class DescentPhaseOptions(AviaryOptionsDictionary):
+
+    def declare_options(self):
+
+        self.declare(
+            'analytic',
+            types=bool,
+            default=False,
+            desc='When set to True, this is an analytic phase.'
+        )
+
+        self.declare(
+            'reserve',
+            types=bool,
+            default=False,
+            desc='Designate this phase as a reserve phase and contributes its fuel burn '
+            'towards the reserve mission fuel requirements. Reserve phases should be '
+            'be placed after all non-reserve phases in the phase_info.'
+        )
+
+        self.declare(
+            name='target_distance',
+            default=None,
+            units='m',
+            desc='The total distance traveled by the aircraft from takeoff to landing '
+            'for the primary mission, not including reserve missions. This value must '
+            'be positive.'
+        )
+
+        self.declare(
+            'target_duration',
+            default=None,
+            units='s',
+            desc='The amount of time taken by this phase added as a constraint.'
+        )
+
+        self.declare(
+            name='fix_initial',
+            types=bool,
+            default=False,
+            desc='Fixes the initial states (mass, distance) and does not allow them to '
+            'change during the optimization.'
+        )
+
+        self.declare(
+            name='input_initial',
+            types=bool,
+            default=False,
+            desc='Links all states to a calculation external to this phase.'
+        )
+
+        self.declare(
+            name='EAS_limit',
+            default=0.0,
+            units='kn',
+            desc='Value for maximum speed constraint in this phase.'
+        )
+
+        self.declare(
+            name='mach_cruise',
```

Comment:
I can't tell in the code if this is just a guess for mach_cruise. Since this is descent, it probably only uses this value for the first point of mach at t_0 for descent.

Reply:
It's an actual boundary constraint, but this is definitely one of those cases where we have to ask, do we really need both "mach_constraint" and "target_mach"? The latter is a boolean switch to activate the former.